src/persistence.ts

```diff
@@ -9,6 +9,7 @@ export interface CreateResourceRequest {
     resourceType: string;
     resource: any;
     id?: string;
+    ttl?: number; // used for archiving

```

Comment:
Maybe we should add in the comment that this value should be timeFromEpoch value in ms. Also do we need to add this field for Update and Patch requests?

Reply:
Thoughts on changing this to `ttlInSeconds` to make it more explicit, what the value should be? By default DDB requires this value to be in seconds, but in JS `Date.now()` returns the time in `ms`.

https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/time-to-live-ttl-before-you-start.html

Reply:
Yeah, I can add a more detailed comment and documentation here. Sorry, it's actually the timestamp in Unix epoch timestamp in seconds, which matches the dynamoDB ttl definition. I will update the comment and documentation to match that of the dynamoDB documentation of ttl here.

I started down the path of updating the ttl on put and patch but didn't want to take a risk the risk of breaking the transaction logic. I figured an absolute expiration as opposed to a sliding expiration would be cleaner right now. It also avoids the non-deterministic scenario where a dynamoDB item has technically expired but the background scanner process hasn't yet picked up the item and started processing yet. Which that time frame looks to be documented at up to 48 hours after a ttl expiration. I'm not super passionate on this though so if you think a sliding expiration makes more sense, I can spend the time to add the logic for put and patch.

Reply:
In general we want to avoid making many small separate changes to the interface, which is why I think it's better to add those fields to PATCH and PUT on the interface now. Since those fields are optional, we leave it to the implementer of the other packages whether they want to take advantage of `ttlInSeconds` in their PATCH or PUT operations. 

Technically, even if we support CREATE only, wouldn't it still be possible that an item that is supposed to have expire isn't deleted till 48 hours later? 

We can set a rule in the persistence package not to accept CREATE, PATCH, or PUT requests with `ttlInSeconds` that are in the past. We will also have an SLA that those items are not guaranteed to be deleted immediately when they expire, but will be deleted within 48 hours.

Reply:
Yeah, this makes a lot of sense on both accounts. I will update the variable name to `ttlInSeconds` and add to the PATCH and PUT interfaces now. I'll also spend some time to go back and try and implement the PATCH and PUT flows to support a sliding expiration. The configuration upstream could then allow someone to set either an absolute or sliding expiration per resource.

Thanks for your help here and quick turnaround.